### ISSUE TITLE
Typescript typings

### ISSUE THREAD

**Issue by organic-scholar** [orchidjs/tom-select#66]
**Describe the bug**
Typescript is unable to recognize typings in present in the package.

**To Reproduce**
in app.ts
`import TomSelect from 'tom-select`

**Expected behavior**
It should be able to resolve type completions.

**Comment by oyejorge** [comment]
Could you provide some more context?

**Comment by oyejorge** [comment]
Haven't heard anything back so I'm gonna close this. If it does still need to be fixed, we can open the issue back up when we have more information.

**Comment by chapterjason** [comment]
@oyejorge We face the same issue. We want only use the typings, but we actually can only use the [`src`](https://github.com/orchidjs/tom-select/blob/master/package.json#L16) which creates some errors if our requirement on browsers is different and not [`esnext`](https://github.com/orchidjs/tom-select/blob/master/.config/tsconfig.json#L7).

In the `package.json` the `types` property should be `./dist/types/tom-select.d.ts` (I think that will be the output? It is not checked in.)

**Comment by oyejorge** [comment]
This is very similar to [a setup a tried earlier](https://github.com/orchidjs/tom-select/commit/e6c846a8771ae81cc61af93a5d064d8e0316fefe#diff-6b1fdfc59614e520241c7ff9ecdee17d0ad5da51a8568be338aaa85b2ca6d463). However, I ran into a problem where it didn't emit the types already defined in /src/types.

However, I may have just found [the missing piece](https://stackoverflow.com/questions/56018167/typescript-does-not-copy-d-ts-files-to-build).

**Comment by oyejorge** [comment]
Ok, should be fixed with 1.7.7

**Comment by marceloverdijk** [comment]
I'm trying to use the TypeScript typings as well, but I'm a bit stuck.

I have some external js theme that automatically creates TomSelect instances so that is fine.

But in some project related ts file I want to access this TomSelect instance like:

```
const $countryTomSelect = $('#country-filter')[0]!!.tomselect
```

and then attach e.g.:

```
$countryTomSelect.on('change', () => applyFilters())
```

this works, but it does not recognise the `tomSelect`... resulting in:
`Property 'tomselect' does not exist on type 'HTMLElement'.`

Is there something I could to use TomSelect typings this way?


====

I for example also use Bootstrap-Table and there I have typings.
e.g.:

```
import type { BootstrapTableOptions } from 'bootstrap-table'

const tableOptions: BootstrapTableOptions = { .. }
$('#table').bootstrapTable(tableOptions)
```

here it recognises the `bootstrapTable` method and typings work
